### PR TITLE
Escape shell args to prevent spaces etc.

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

File: lib/WatcherProcess.php [hunk 1]
```diff
@@ -248,10 +249,11 @@ private function onDeadIpcClient(string $readWatcherId, $ipcClient) {
...
         $parts[] = \PHP_BINARY;
         if ($ini = \get_cfg_var("cfg_file_path")) {
-            $parts[] = "-c \"{$ini}\"";
+            $ini = escapeshellarg($ini);
+            $parts[] = "-c {$ini}";
         }
         $parts[] = "-d zend.assertions=" . ini_get("zend.assertions");
-        $parts[] = __DIR__ . "/../bin/aerys-worker";
+        $parts[] = escapeshellarg(__DIR__ . "/../bin/aerys-worker");
         $parts[] = "-i {$this->ipcServerUri}";
```
Comment (staabm) [inline lib/WatcherProcess.php]: what about this var?

File: lib/WatcherProcess.php [hunk 2]
```diff
@@ -248,10 +249,11 @@ private function onDeadIpcClient(string $readWatcherId, $ipcClient) {
...
         if ($console->isArgDefined("config")) {
             $parts[] = "-c " . $console->getArg("config");
```
Comment (staabm) [inline lib/WatcherProcess.php]: are those console-args "secure-by-default"?

File: lib/WatcherProcess.php (hunk 1 above)
Comment (kelunik) [inline lib/WatcherProcess.php]: Shouldn't be a problem, but may be a problem when switched to unix domain sockets and spaces in file paths.

File: lib/WatcherProcess.php (hunk 2 above)
Comment (kelunik) [inline lib/WatcherProcess.php]: Should be, but anyway. Added escaping for all args.

**Comment by staabm** [comment]
after all its also a security fix

**Comment by kelunik** [comment]
We need a security contact address for real security issues.

**Comment by staabm** [comment]
> We need a security contact address for real security issues.

agree. this should also be documented on the website and github repos on how-to report security related issues.

**Comment by kelunik** [comment]
See #48.

**Comment by bwoebi** [comment]
Merged as 7bec44a with a few improvements

**Pull request closed without merging**